controller/workitem.go

```diff
@@ -561,6 +565,252 @@ func getVersion(version interface{}) (int, error) {
+	}
+	// done, return serialized result.
+	return buf.String(), nil
+}
+
+// extractWorkItemTypeFields extracts the field information for a wit; it returns slices for
+// field labels and field keys
+func extractWorkItemTypeFields(wit workitem.WorkItemType) ([]string, []string, error) {
+	fieldLabels := []string{}
+	fieldKeys := []string{}
+	for fieldKey, fieldDefinition := range wit.Fields {
+		// extract the label and key
+		fieldLabels = append(fieldLabels, fieldDefinition.Label)
+		fieldKeys = append(fieldKeys, fieldKey)
+	}
+	return fieldLabels, fieldKeys, nil
+}
+
+// convertWorkItemFieldValues extracts and converts the wi field values; it returns a map
+// that maps field keys to converted field values
+func convertWorkItemFieldValues(ctx context.Context, app application.Application, uuidStringCache *map[string]string, wit workitem.WorkItemType, wi workitem.WorkItem) (map[string]string, error) {
+	fieldMap := make(map[string]string)
+	for fieldKey, fieldDefinition := range wit.Fields {
+		// convert the value to a string for the CSV
+		fieldValueGeneric := wi.Fields[fieldKey]
+		fieldType := fieldDefinition.Type
+		fieldValueStrSlice, err := fieldType.ConvertToStringSlice(fieldValueGeneric)
+		if err != nil {
+			return nil, errs.Wrapf(err, "failed to convert type value to string for field key: %s", fieldKey)
+		}
+		var convertedValue *string
+		// now retrieve and, if needed, resolve the id value.
+		switch fieldType.(type) {
+		case workitem.ListType:
+			var converted string
+			kind := fieldType.(workitem.ListType).ComponentType.Kind
+			delim := ""
+			for _, elem := range fieldValueStrSlice {
+				elemConvertedValue, err := convertValueToString(ctx, app, uuidStringCache, fieldValueGeneric, []string{elem}, fieldKey, kind)
+				if err != nil {
+					return nil, errs.Wrapf(err, "failed to convert compound type value to string for field key: %s", fieldKey)
+				}
+				converted = converted + delim + *elemConvertedValue
+				delim = ";"
+			}
+			convertedValue = ptr.String(converted)
+		case workitem.EnumType:
+			kind := fieldType.(workitem.EnumType).BaseType.Kind
+			convertedValue, err = convertValueToString(ctx, app, uuidStringCache, fieldValueGeneric, fieldValueStrSlice, fieldKey, kind)
+		default:
+			// all other Kinds don't need compound resolving.
+			kind := fieldType.GetKind()
+			convertedValue, err = convertValueToString(ctx, app, uuidStringCache, fieldValueGeneric, fieldValueStrSlice, fieldKey, kind)
+		}
+		if err != nil {
+			return nil, errs.Wrapf(err, "failed to resolve type value to string for field key: %s", fieldKey)
+		}
+		fieldMap[fieldKey] = *convertedValue
+	}
+	return fieldMap, nil
+}
+
+// convertValueToString converts a value to a string. This includes ID resolving if needed.
+func convertValueToString(ctx context.Context, app application.Application, uuidStringCache *map[string]string, fieldValueGeneric interface{}, fieldValueStrSlice []string, fieldKey string, kind workitem.Kind) (*string, error) {
```

Comment:
I just noticed this, but why does this function return a pointer to a string? Can't we just return an empty string in case of an error? That's usually how its done.